### DISCUSSION TITLE
`next run <script>`

### DISCUSSION THREAD

**Discussion by brandonbloom** [vercel/next.js#23214] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

I'd like a way to run scripts for general maintenance tasks with zero config and utilizing all of the Next.js machinery (transpilation, module aliasing, polyfills, configuration loading, etc).

## Additional context

See https://github.com/vercel/next.js/pull/15276#issuecomment-660336199

**Comment by nzjoel1234** [comment]
This would be helpful for me as I want to build a [Lunr ](https://lunrjs.com/) search index at build time and store it as a JSON file in `/public`.

Ideally I would like to use typescript for this.

I am currently running my script from within `getStaticProps` but this feels a bit like a hack.
It would be nice if there was a more suitable place to hook in scripts like this.
